std/file.d

```diff
@@ -1405,6 +1405,13 @@ assert("/usr/share/include".isDir);
     }
 }
 
+@property bool isDir(R)(auto ref R name)
+if (!(isInputRange!R && isSomeChar!(ElementEncodingType!R))
+    && is(StringTypeOf!R))
+{
+    return isDir(cast(StringTypeOf!R)name);
```

Comment:
When you explicitly instantiate `isDir!(StringTypeOf!R)(name)` you about the cast and prolly allow this function to be `@safe`.